### PR TITLE
chore: onboard Azure CNI images to MCR

### PR DESCRIPTION
**Pull request by saiyan86**
**Reason for Change**:

This PR changes Azure CNI image references to mcr.microsoft.com

**Issue Fixed**:

N/A

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Notes**:
N/A

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
commit message needs to use semantic message

/hold

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ var (
 		TillerImageBase:                  "gcr.io/kubernetes-helm/",
 		ACIConnectorImageBase:            "microsoft/",
 		NVIDIAImageBase:                  "nvidia/",
-		AzureCNIImageBase:                "containernetworking/",
+		AzureCNIImageBase:                "mcr.microsoft.com/containernetworking/",
```
Comment (CecileRobertMichon) [inline pkg/api/azenvtypes.go]: is this available in Mooncake and other sovereign clouds?

Comment (saiyan86) [reply pkg/api/azenvtypes.go]: Not yet, it's being rolled out.

Comment (CecileRobertMichon) [reply pkg/api/azenvtypes.go]: So this PR shouldn't be merged until it's rolled out?

Comment (saiyan86) [reply pkg/api/azenvtypes.go]: I think it's OK to merge. https://github.com/Azure/aks-engine/blob/f518126807a5f05ed7633eac302fb52a496e8213/pkg/api/azenvtypes.go#L233 has the Mooncake specific URL.

**Comment by CecileRobertMichon** [comment]
@saiyan86 FYI I rebased and forced pushed to resolve the merge conflict

**Comment by saiyan86** [comment]
> @saiyan86 FYI I rebased and forced pushed to resolve the merge conflict

Thanks for this!

**Pull request merged**